### PR TITLE
[network] https tracer tests

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

Add more https tests based on curl and wget using openssl guntls ...

### Motivation

Validating https on our CI

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/functional_test/system_probe.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ kitchen_test_system_probe_linux_x64:
   before_script:
     - cd $DD_AGENT_TESTING_DIR
     - bash -l tasks/kitchen_setup.sh
+    - source /root/.bashrc && conda activate ddpy3 && conda install openssl gnutls
```
Comment (mx-psi) [inline .gitlab/functional_test/system_probe.yml:34]: Unless there is a very good reason, we should not modify the conda environments on the Gitlab jobs, this should be modified over at https://github.com/DataDog/datadog-agent-buildimages. 

If these changes are not related to the Conda environment we prefer installing via the package manager

File: pkg/network/http/http_stats.go [hunk 2]
```diff
@@ -113,6 +113,19 @@ func NewKeyTuple(saddr, daddr util.Address, sport, dport uint16) KeyTuple {
 	}
 }
 
+// L3L4Equal return true is both KeyTuple are equal
+func (t *KeyTuple) L3L4Equal(t2 KeyTuple) bool {
```
Comment (brycekahle) [inline pkg/network/http/http_stats.go:117]: I don't see this used anywhere.

File: .gitlab/functional_test/system_probe.yml (hunk 1 above)
Comment (brycekahle) [reply .gitlab/functional_test/system_probe.yml:34]: @nplanel can you install the packages some other way?
